### PR TITLE
Add upgrade docs for users of `@kitconcept/volto-blocks-grid` addon

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

**Review by davisagli — COMMENTED** [review]
Looks accurate from a technical perspective, leaving it to @stevepiercy for docs review

File: docs/source/upgrade-guide/index.md [hunk 1]
```diff
@@ -265,6 +265,37 @@ It requires the latest version of `plone.restapi` (>=8.42.0) installed in the ba
...
 You can consider removing it if you were shadowing it in your project.
 
+(grid-block-migration-from-kitconcept-volto-blocks-grid-label)=
+
+### Grid block migration from `@kitconcept/volto-blocks-grid`
+
+The grid block was added to Volto in version 17.0.0-alpha.16.
+It is based on the `@kitconcept/volto-blocks-grid` add-on version 7.x.x.
+
+If you previously used `@kitconcept/volto-blocks-grid` in your project based on a Volto version before 17.0.0-alpha.16, then your site will show two available block types both named `Grid`.
```
Comment (stevepiercy) [inline docs/source/upgrade-guide/index.md:275]: @sneridagh @davisagli can you verify this sentence for accuracy?

**Review by stevepiercy — COMMENTED** [review]
I deduplicated content and did a slight rewrite. I have one question in my review. Otherwise this is good to go, and I approve.

File: docs/source/upgrade-guide/index.md (hunk 1 above)
Comment (sneridagh) [reply docs/source/upgrade-guide/index.md:275]: It is accurate, then later it explains how to deal with that situation, by using `restricted` property.

Comment (sneridagh) [reply docs/source/upgrade-guide/index.md:275]: Merging!

**Pull request merged**